x-pack/plugin/kql/src/main/java/org/elasticsearch/xpack/kql/parser/ParsingException.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.kql.parser;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.rest.RestStatus;
+
+import static org.elasticsearch.common.logging.LoggerMessageFormat.format;
+
+public class ParsingException extends ElasticsearchException {
```

Comment:
ℹ️ Using our own exception type for KQL parsing errors.

Reply:
Should this be named something explicit like `KqlParsingException`?

Reply:
Done.